scheduler/scheduler.go

```diff
@@ -102,6 +100,8 @@ func (s *Service) StartScheduler(versionedcfg *v1beta2config.KubeSchedulerConfig
 		return xerrors.Errorf("create scheduler: %w", err)
 	}
 
+	s.currentSchedulerCfg = converted.DeepCopy()
```

Comment:
This `converted` Scheduler Configuration contains the internal plugins for simulator. (because of `convertConfigurationForSimulator`. Described [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/61eebb3e302e8960bcce804e859510c0af71489a/scheduler/scheduler.go#L128).)

So, in this implementation, users can see internal `XXXXXXForSimulator` plugins in SchedulerConfiguration on Web UI. 
![スクリーンショット 2022-03-03 21 43 45](https://user-images.githubusercontent.com/44139130/156567067-5ed42a8e-6284-444a-b470-30a8eda54cdc.png)

From this reason, `s.currentSchedulerCfg` should be updated before `convertConfigurationForSimulator` like previous implementation. 

You changed `convertConfigurationForSimulator` to return versioned config not to allow users to overwrite version, right? So, it may be simpler to overwrite version in [`ApplySchedulerConfig`](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/61eebb3e302e8960bcce804e859510c0af71489a/server/handler/schedulerconfig.go#L29) instead of modifing version in `convertConfigurationForSimulator`. (yeah, there are other possible solutions to this problem, I'll leave it to you.)

Reply:
@sanposhiho Thank you for catching that and suggesting ApplyConfig.  Yep!  Next commit is much simpler to only apply the profiles and make sure the other changes don't re-occur in the UI.  It really seems to be lacking validation, but I think that will make more sense when I add beta3 for a different PR/issue.  I didn't squash in case you'll want to see both commits, but after it's squashed much of the first will be gone.